### PR TITLE
Refactor autocomplete

### LINKED ISSUE #2283: generic_work is hardcoded in edit_metadata.js
This may cause batch_upload not to have autocomplete. Additionally autocomplete may not work unless the work is GenericWork.

### LINKED ISSUE #2287: Location authorities are broken
```
Started GET "/authorities/generic_works/location?q=Saint" for ::1 at 2016-06-24 10:14:52 -0700
Processing by AuthoritiesController#query as JSON
Parameters: {"q"=>"Saint", "model"=>"generic_works", "term"=>"location"}
Completed 500 Internal Server Error in 308ms (ActiveRecord: 0.0ms)

Module::DelegationError - ActiveResource::Collection#each delegated to to_a.each, but to_a is nil: nil:
  activeresource (4.1.0) lib/active_resource/collection.rb:7:in `rescue in each'
  activeresource (4.1.0) lib/active_resource/collection.rb:7:in `each'
  activeresource (4.1.0) lib/active_resource/collection.rb:67:in `collect!'
  activeresource (4.1.0) lib/active_resource/base.rb:1043:in `instantiate_collection'
  /Users/jcoyne/workspace/sufia/app/models/geo_names_resource.rb:11:in `instantiate_collection'
  activeresource (4.1.0) lib/active_resource/base.rb:1012:in `find_every'
  activeresource (4.1.0) lib/active_resource/base.rb:919:in `find'
  /Users/jcoyne/workspace/sufia/app/models/geo_names_resource.rb:16:in `find_location'
  /Users/jcoyne/workspace/sufia/app/controllers/authorities_controller.rb:9:in `query'
```

### PR DESCRIPTION
**Pull request by jcoyne**
@projecthydra/sufia-code-reviewers

Makes the following changes:
1) don't hard code urls or "generic_work" in the javascript
2) convert to es6
3) The location autocomplete was using one method for the first field,
and a different method for subsequent fields.  This makes them use the
same method each time.
4) Use questioning authority for authority lookups.

Fixes #2283
Fixes #2287

### CODE REVIEW

**Comment by jcoyne** [comment]
I'm aware that the jasmine tests are failing, but pushing this now as I'm about to hop on a plane.

File: app/assets/javascripts/sufia/autocomplete.es6 [hunk 1]
```diff
@@ -0,0 +1,51 @@
+export class Autocomplete {
+  constructor() {
+  }
+
+  // This is the initial setup for the form.
+  setup() {
+      $('[data-autocomplete]').each((index, value) => {
+          let selector = $(value)
+          switch (selector.data('autocomplete')) {
```
Comment (cam156) [inline app/assets/javascripts/sufia/autocomplete.es6]: @jcoyne Seems like this code and the code at line 25 are very similar. Wondering if they could be combined so we are not testing for the same hard coded strings in two places.  This is not a show stopper for the PR by any means...

File: app/assets/javascripts/sufia/autocomplete/language.es6 [hunk 2]
```diff
@@ -0,0 +1,25 @@
+export class Language {
```
Comment (cam156) [inline app/assets/javascripts/sufia/autocomplete/language.es6]: @jcoyn The 3 classes Language, Subject &  look to be exact copies.  Was that intentional?

File: config/routes.rb [hunk 3]
```diff
@@ -118,9 +118,6 @@
   # advanced routes for advanced search
   get 'search' => 'advanced#index', as: :advanced
 
-  # Authority vocabulary queries route
```
Comment (cam156) [inline config/routes.rb]: @jcoyne I'm surprised there is no route for searching the Authorities needed.  Is this coming in from Questioning Authority?

Comment (cam156) [inline config/routes.rb]: @jcoyne Nevermind I see it in the generators...

File: sufia.gemspec [hunk 4]
```diff
@@ -59,7 +59,7 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency "factory_girl_rails", '~> 4.4'
   spec.add_development_dependency "equivalent-xml", '~> 0.5'
   spec.add_development_dependency "jasmine", '~> 2.3'
-  spec.add_development_dependency 'rubocop', '~> 0.40'
+  spec.add_development_dependency 'rubocop', '~> 0.40.0'
```
Comment (cam156) [inline sufia.gemspec]: 👏

File: spec/routing/route_spec.rb [hunk 5]
```diff
@@ -115,12 +115,6 @@
     end
   end
 
-  describe 'Authorities' do
```
Comment (cam156) [inline spec/routing/route_spec.rb]: A Test for the QA route generated?

Comment (jcoyne) [inline spec/routing/route_spec.rb]: That's already tested in QA. Does it make sense to test it again?

File: app/assets/javascripts/sufia/autocomplete/language.es6 (hunk 2 above)
Comment (jcoyne) [inline app/assets/javascripts/sufia/autocomplete/language.es6]: yes, there's room for refactoring here, but I wanted to leave open the possibility of customizing the options per class.

File: spec/routing/route_spec.rb (hunk 5 above)
Comment (cam156) [inline spec/routing/route_spec.rb]: @jcoyne yes and no:  this was the line I was thinking needed a test: https://github.com/projecthydra/sufia/pull/2288/files#diff-25c1ec9f5b287e7f6a20f696507ab602R24
There may be a better test than a route test to make sure the engine was mounted.

**Pull request merged**